### DISCUSSION TITLE
Public API is now in production 🎉

### DISCUSSION THREAD

**Discussion by mehdyouras** [iopool/community#1] [Announcements]
# Documentation
https://help.iopool.com/fr/articles/5537423-api-publique-iopool

**Comment by john-does-it** [comment]
Yeah boi! 🗡️

**Comment by Samuel-Samuel** [comment]
Super !

**Comment by Tic57** [comment]
vraiment hâte de tester ca

**Comment by Septain21** [comment]
Bonjour

J’ai tenté depuis des jours de faire fonctionner l’API REST de iopool sans succès.

Voici mon flow :

```
[{"id":"eedf1a193d6f4345","type":"tab","label":"iopool","disabled":false,"info":"","env":[]},{"id":"a28c796f940d4ff8","type":"http` request","z":"eedf1a193d6f4345","name":"","method":"GET","ret":"obj","paytoqs":"ignore","url":"https://api.iopool.com/v1/pool/xxxxxxxxx","tls":"","persist":false,"proxy":"","authType":"basic","senderr":false,"headers":[],"x":530,"y":380,"wires":[["448df951617d97dc","293353468e673357","647b97e0ec7fe810","edf6b8f31568d7ca"]]},{"id":"86736107966c7970","type":"inject","z":"eedf1a193d6f4345","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":200,"y":380,"wires":[["ce112331556eb3be"]]},{"id":"ce112331556eb3be","type":"function","z":"eedf1a193d6f4345","name":"","func":"msg.headers = { \"x-api-key\": \"xxxxxxxxxxxxxx\"};\nreturn msg;","outputs":1,"noerr":0,"initialize":"// Code added here will be run once\n// whenever the node is started","finalize":"","libs":[],"x":360,"y":380,"wires":[["a28c796f940d4ff8"]]},{"id":"448df951617d97dc","type":"function","z":"eedf1a193d6f4345","name":"temperature","func":"msg.payload = msg.payload.latestMeasure.temperature\nmsg.topic = \"piscine/temperature\"\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":810,"y":260,"wires":[["08d1af35247fb751","82cc55453e99221f"]]},{"id":"08d1af35247fb751","type":"mqtt out","z":"eedf1a193d6f4345","name":"","topic":"","qos":"","retain":"","respTopic":"","contentType":"","userProps":"","correl":"","expiry":"","broker":"f14d628b09a34bc3","x":1290,"y":340,"wires":[]},{"id":"293353468e673357","type":"function","z":"eedf1a193d6f4345","name":"ph","func":"msg.payload = msg.payload.latestMeasure.ph\nmsg.topic = \"piscine/ph\"\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":790,"y":300,"wires":[["08d1af35247fb751","82cc55453e99221f"]]},{"id":"647b97e0ec7fe810","type":"function","z":"eedf1a193d6f4345","name":"capacite_de_desinfection","func":"msg.payload = msg.payload.latestMeasure.orp\nmsg.topic = \"piscine/orp\"\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":850,"y":340,"wires":[["08d1af35247fb751","82cc55453e99221f"]]},{"id":"edf6b8f31568d7ca","type":"function","z":"eedf1a193d6f4345","name":"date_de_derniere_mesure","func":"msg.payload = msg.payload.latestMeasure.measuredAt\nmsg.topic = \"piscine/date\"\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":850,"y":380,"wires":[["08d1af35247fb751","82cc55453e99221f","107a6b82df05e266"]]},{"id":"82cc55453e99221f","type":"debug","z":"eedf1a193d6f4345","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":1130,"y":260,"wires":[]},{"id":"107a6b82df05e266","type":"moment","z":"eedf1a193d6f4345","name":"","topic":"","input":"payload","inputType":"msg","inTz":"Europe/Brussels","adjAmount":0,"adjType":"days","adjDir":"add","format":"D-MMMM-YYYY HH:mm","locale":"fr","output":"payload","outputType":"msg","outTz":"Europe/Brussels","x":1100,"y":400,"wires":[["08d1af35247fb751"]]},{"id":"678851ddd30a1192","type":"configurable interval","z":"eedf1a193d6f4345","name":"Every 15 min","interval":"15","onstart":false,"do_enable":true,"msg":"ping","showstatus":true,"unit":"minutes","statusformat":"YYYY-MM-D HH:mm:ss","x":170,"y":300,"wires":[["ce112331556eb3be"]]},{"id":"f14d628b09a34bc3","type":"mqtt-broker","name":"","broker":"192.168.0.146","port":"1883","clientid":"","autoConnect":true,"usetls":false,"protocolVersion":"3","keepalive":"60","cleansession":true,"birthTopic":"","birthQos":"0","birthRetain":"false","birthPayload":"","birthMsg":{},"closeTopic":"","closeQos":"0","closeRetain":"false","closePayload":"","closeMsg":{},"willTopic":"","willQos":"0","willRetain":"false","willPayload":"","willMsg":{},"userProps":"","sessionExpiry":""}]
```

Une idée de l’endroit où ca pêche ?

Merci

**Reply by Jeoffreybauvin** [reply]
Bonjour @Septain21 

Je ne pense pas qu'iopool s'amuse à importer votre flow node-red pour le debug :D. De mon coté, j'utilise également Node-Red pour l'api d'iopool et ça marche bien. Je vous suggère soit de poster l'erreur que vous rencontrez, soit d'expliquer un peu plus votre souci...

![image](https://user-images.githubusercontent.com/701928/182197522-67a652c6-ad8c-42cb-8d67-42eaa805e8a8.png)

**Reply by Septain21** [reply]
Bonsoir Jeoffrey 

Effectivement le débug ne fait pas partie du travail de SAV d'IOPOOL mais cela peut aider voir inciter certains à passer le pas.

Voici le retour d'erreur sur nodered : 
TypeError: Cannot read properties of undefined (reading 'temperature')

Merci de ton aide

**Reply by Jeoffreybauvin** [reply]
J'ai tenté d'importer ton flow, mais il y a certains nodes que je n'utilise pas et que je connais pas, donc aucune idée.

**Reply by Septain21** [reply]
Ah c’est bizarre parce que j’ai copié votre flow je crois directement du forum hacf.fr

**Reply by jeff0077** [reply]
Bonjour, avez vous réussi au final car je suis intéressé par votre flow si c'est le cas

**Comment by apwelsh** [comment]
Any chance of getting a local API option for those of us that use home automation systems that do not use the internet?

**Comment by didgonzalez** [comment]
Bonjour, 
je débute dans home assistant et je suis vraiment débutant. je souhaite integrer ma sonde iopool dans mon dasboard, mais je ne sais comment faire. si quelqu'un a un tuto ou du temps pour m'expliquer.
cdt
didier

**Reply by SebastienBeaumont** [reply]
Hello il existe une discussion spécifique sur Home Assistant ici : #14 
J'espère que cela pourra vous aider!

**Comment by didgonzalez** [comment]
BjrNon pas réussi Cordialement,Didier GonzalezLe 19 nov. 2022 à 16:08, jeff0077 ***@***.***> a écrit :﻿
Bonjour, avez vous réussi au final car je suis intéressé par votre flow si c'est le cas

**Comment by mguyard** [comment]
Bonjour,

Pourriez-vous mettre à jour la doc API avec les changements qui ont était effectués ?
J'ai identifié par exemple le mode ACTIVE_WINTER qui n'est pas dans la doc. Peut être existe t'il d'autres changements.

Merci

**Reply by mehdyouras** [reply]
Bonjour @mguyard, en effet le mode `ACTIVE_WINTER` a été ajouté il correspond au mode hivernage actif dans l'application. Je peux confirmer que c'est la seule disparité avec la documentation. Je vais tâcher de la mettre à jour bientôt. Merci pour vos retours !

**Comment by didgonzalez** [comment]
BonjourComment faire svpCordialement,Didier GonzalezLe 29 mai 2023 à 15:27, Marc GUYARD ***@***.***> a écrit :﻿
Bonjour,
Pourriez-vous mettre à jour la doc API avec les changements qui ont était effectués ?
J'ai identifié par exemple le mode ACTIVE_WINTER qui n'est pas dans la doc. Peut être existe t'il d'autres changements.
Merci

**Reply by Jeoffreybauvin** [reply]
Le message de @mguyard  s'adressait à l'équipe iopool :)

**Comment by StenGarny** [comment]
@mehdyouras est-il possible de récupérer le taux de sel de la version Salt ?
Merci d'avance !
Sten
